<a href="https://colab.research.google.com/github/gingyy/adopt-a-stream-georgia/blob/main/data_dictionary_version_2_25.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
from google.colab import files
uploaded = files.upload()

Saving export_dataframe_stage2.csv to export_dataframe_stage2.csv


In [4]:
import io
import pandas as pd
df2 = pd.read_csv(io.BytesIO(uploaded['export_dataframe_stage2.csv'])) #replace with file name

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (14,20,21,23,24,31,32,33,35,36,56,63,64,65,72,73,74,77,78,79,86,87,88) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [52]:
df2.head()

group_rid  ...                            wbd
0        840  ...         West Fork Little River
1        840  ...         West Fork Little River
2        297  ...  Turkey Creek-Appalachee River
3        297  ...  Turkey Creek-Appalachee River
4        297  ...  Turkey Creek-Appalachee River

[5 rows x 90 columns]

# group_rid

Numerical identifier associated with ```GroupName```
* There are 992 distinct group_rid


# GroupName

Name of citizen group who collected the associated data entry
* 992 distinct group names

# site_rid

Numerical identifier associated with ```SiteLocation```
* 2637 site ids
* Will further investigate if it corresponds to any geographical factors

# event_rid

Numerical identifier associated with ```event_date```
* Effectively an index for each entry 
* Does not appear to be chronological
* Unique value for each data entry
* Appears to skips values at random
* 59598 event ids
* Will further investigate why it skips over some values

# Comments

Notes describing the condition of the river (trash, etc.)
* 94.6% of data in this column are NaN 
* 3163 non-null Comments, 2725 of unique ones
* All kinds of environmental features noticed, including vegetation, water quality, water level, smell, trash ... or just used as a place of communication between researchers

# comments

Notes with scientific information (bacteria, etc.)
* 99% are NaN values
* 465 unique comments out of total 541 non-null
* more focused on the detection and abnormal features of bacteria level

# warnings

Standardized labels all refer to "Bacterial Warnings:"
* 99.5% are NaN values
* In total 165 groups reported warnings, and all of them reported the same thing
* Some groups have more warnings than others (EX. group#2986 has 23 warnings and group#1398 - 12 warnings, group#6796&316 - 10 warnings, group#2858&6567 - 9 warnings, group#6594 - 8 warnings, group#6649 - 7 warnings, group#2388&4055&2500 - 6 warnings, all others are below 5)

# SiteName

Name of the body of water or general location
* 1892 site names
* Directly associated with columns ```site_rid```,	```state```,	```county```,	```lat```,	```long```,	```SiteName```,	```SiteLocation```
* Appears to be a more general label for ```wbd```

# SiteLocation

Coordinates of the site location
* Direct combination of columns ```lat``` and ```long```, in that order
* Coordinates are separated by two spaces between the latitute and longitude

# event_date

Date of the collection event in format YYYY-MM-DD HH:MM:SS	
* Partially associated with ```event_rid```
* ```event_rid``` contains all unique values while ```event_date``` contains 8356 non-unique values

# volunteer_time

[Units = Minutes] Number of minutes spent at site by volunteers
* Has a mean of 101 minutes, standard deviation of 657 minutes
* Has a minimum of 0 minutes, and has 23056 NaN values
* Has a maximum of 120210 minutes, which equates to 83.5 days
* The middle 50% lies between 45 minutes and 120 minutes

# createdby

Contact information for party associated with the data entry
* 78% of entries are "EPDMIG"
* "EPDMIG" rows have the same several ```createddate``` values
* Other values are emails, with a combination of personal, academic institutional, and organizational addresses

# createddate

Date the data was uploaded, in format M/D/YY HH:MM AM/PM

# data_entry

An indicator or label corresponding to ```group_rid``` and ```GroupName``` but with more NA values
* 1574 distinct values
* Float data type but all values end in .0, used as a categorical label
* Includes inconsistencies with different ```data_entry``` values for the same ```group_rid``` and ```GroupName```
* Often, same ```data_entry``` corresponds to same ```site_rid``` with a few exceptions, in which one ```data_entry``` corresponds to several consecutive ```site_rid```. (EX. ```data_entry``` 3307.0 to ```site_rid``` 45,46,47)


# participants

[Units = Number of People] Number of people involved in the data collection event
* Minimum value of 0 
* Zero NaN values
* Middle 50% range is 1-2 people
* Maximum of 111 people